File: tests/journal.lib/debug_empty.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// verify that empty messages are handled correctly
int main() {
    // make a debug channel
    pyre::journal::debug_t channel("tests.journal.debug");
    // activate the channel
    channel.activate();
    // but send the output to the trash
    channel.device<pyre::journal::trash_t>();

    // inject nothing
    channel << pyre::journal::endl;

    // all done
    return 0;
}


// end of file

File: tests/journal.ext/debug_empty.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that injecting an empty message is handled correctly
    """
    # get the channel
    from journal.ext.journal import Debug

    # make a channel
    channel = Debug(name="test.journal.debug")
    # activate it
    channel.activate()

    # inject an empty message
    channel.log()

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
